django_airavata/apps/groups/views.py

```diff
@@ -2,22 +2,174 @@
 from django.conf import settings
 from django.contrib.auth.decorators import login_required
 from django.shortcuts import render, redirect
+from .forms import CreateForm, AddForm, RemoveForm
+from django.contrib import messages
+from apache.airavata.model.sharing.ttypes import UserGroup
+from apache.airavata.model.sharing.ttypes import GroupCardinality
+import datetime
+import uuid
 
 import logging
 
 logger = logging.getLogger(__name__)
 
+@login_required
+def groups_manage(request):
+
+    gateway_id = settings.GATEWAY_ID
+    username = request.user.username
+
+    try:
+        owner_list = request.sharing_client.getGroups(gateway_id, 0, -1)
+        owner = []
+        for group in owner_list:
+            if group.ownerId == username:
+                owner.append(group)
+        member_list = request.sharing_client.getAllMemberGroupsForUser(gateway_id, username)
+        member = []
+        for group in member_list:
+            if group.ownerId != username:
+                member.append(group)
+        return render(request, 'django_airavata_groups/groups_manage.html', {
+            'owner': owner, 'member': member
+        })
+    except:
+        logger.exception("Failed to load the Manage Groups page")
+        return redirect('/')
 
 @login_required
 def groups_list(request):
 
     gateway_id = settings.GATEWAY_ID
 
     try:
-        groups = request.sharing_client.getGroups(gateway_id, 0, -1)
+        group_list = request.sharing_client.getGroups(gateway_id, 0, -1)
+        for group in group_list:
+            if group.groupCardinality == 0:
```

Comment:
Looks like you are removing single user groups here but I don't think the app uses groups_list any longer. Maybe remove the groups_list function but move this groupCardinality check to the groups_manage function.